### PR TITLE
Add compaction complete listerner for entire work flow

### PR DESCRIPTION
**Pull request by yukuai518**
1. Made method signature change and constructor change for compaction completion listener
2. Made attributes public in TimeBasedSubDirDatasetsFinder

### CODE REVIEW

File: gobblin-compaction/src/main/java/gobblin/compaction/listeners/SimpleCompactorCompletionListener.java [hunk 1]
```diff
@@ -1,24 +1,29 @@
...
+import org.joda.time.DateTime;
 
 import gobblin.annotation.Alias;
-import gobblin.compaction.conditions.RecompactionCombineCondition;
-import gobblin.compaction.mapreduce.MRCompactor;
+import gobblin.configuration.State;
+import gobblin.compaction.dataset.Dataset;
 
 
 public class SimpleCompactorCompletionListener implements CompactorCompletionListener {
```
Comment (pcadabam) [inline gobblin-compaction/src/main/java/gobblin/compaction/listeners/SimpleCompactorCompletionListener.java]: Add java doc and license

File: gobblin-compaction/src/main/java/gobblin/compaction/listeners/CompactorCompletionListener.java [hunk 2]
```diff
@@ -12,10 +12,11 @@
...
-
-import gobblin.compaction.mapreduce.MRCompactor;
+import java.util.Set;
+import org.joda.time.DateTime;
+import gobblin.compaction.dataset.Dataset;
 
 
 public interface CompactorCompletionListener {
-  void onCompactionCompletion(MRCompactor compactor);
+  void onCompactionCompletion (DateTime intializeTime, Set<Dataset> datasets);
```
Comment (pcadabam) [inline gobblin-compaction/src/main/java/gobblin/compaction/listeners/CompactorCompletionListener.java]: Why do we need `initializeTime`? I think having `MRCompactor` in the signature is useful. Can't you get `datasets` from `MRCompactor`?

File: gobblin-compaction/src/main/java/gobblin/compaction/listeners/SimpleCompactorCompletionListener.java (hunk 1 above)
Comment (yukuai518) [inline gobblin-compaction/src/main/java/gobblin/compaction/listeners/SimpleCompactorCompletionListener.java]: Done

File: gobblin-compaction/src/main/java/gobblin/compaction/listeners/CompactorCompletionListener.java (hunk 2 above)
Comment (yukuai518) [inline gobblin-compaction/src/main/java/gobblin/compaction/listeners/CompactorCompletionListener.java]: Change back to MRCompactor.

**Pull request merged**